.github/workflows/build.yml

```diff
@@ -54,6 +54,13 @@ jobs:
           docker run --rm -idt -p 5000:5000 rigetti/qvm -S
           docker run --rm -idt -p 5555:5555 rigetti/quilc -R
 
+      - name: Fetch Jupyter cache
+        uses: actions/cache@v3
+        with:
+          path: ./docs/build/.jupyter_cache
+          key: jupyter-cache-${{ github.run_id }}
+          restore-keys: jupyter-cache
```

Comment:
1. I'm not sure, is `restore-keys` required? If so, I think it might be missing a `-` at the end to match `key`?
2. My understanding of `key` is it is the thing that will control the use of a cache, so if the key changes, a new cache will be requested. Should we have the key be some sort of hash of the mitiq python files? It's done this way [here](https://github.com/lesteve/scikit-learn-mooc/blob/7243c2a8dfd5e1704986e7b9564c1d0111e9a334/.github/workflows/deploy-gh-pages.yml#L39).

Reply:
1. It makes sense only when you want to make a priority list of keys you want to restore, which is not our case. I don't think it was missing the `-` anyway, since it's just supposed to be a matching prefix, and `jupyter-cache` would match `jupyter-cache-${{ github.run_id }}`. Anyway, it's not required, I will remove it.

2. The key I picked is static as `github.run_id` effectively is the same for each workflow run. Here is my rationale for picking a static key: we are not using this action to create a cache mechanism, which is instead already provided by jupyter cache. All we want to do here is load the jupyter cache, and then sphinx/myst_nb will know how to handle that cache, the same way it happens locally when you run two times `make docs`. If we have a key that is a hash of the mitiq python files, or a hash of all mitiq files, then the cache is invalidated on each PR, which is not what we want. I know it's a bit counterintuitive, and that's because this is a bit of an unorthdox use of this action. We are using it to store and load an existing caching mechanism, rather than for implementing a caching mechanism.